Accessibility Section Revamp (#2661)
telerik/blazor-docs: accessibility/overview.md

slug: accessibility-overview
tags: telerik,blazor,accessibility,overview
published: True
position: 1
previous_url: /accessibility/keyboard-navigation,/accessibility/accessibility-swatch
---

# Blazor Accessibility Overview

Web applications are accessible when they provide control over their features to users with physical or situational disabilities. These disabilities can include various categories, for example, user who are:

> * Unable to perceive content visually or auditory;

Reviewer: ```suggestion
* Unable to perceive content visually or auditory.
```

The keyboard support follows the normal flow of the web page content. Use the `Tab` key to focus a component and then use specific keyboard shortcuts to trigger specific actions. For example, use the arrow keys to move across cells in the Grid or hit `Enter` to invoke a button click.

Most components represent a single tab stop. Once users reach and focus a component, they can leave it with a single `Tab` key press. If the component is more complex, users can walk through its inner elements with the arrow keys, for example, Grid cells, Menu items, Toolbar buttons. Some complex components can accommodate multiple other components. For example, the Grid can host a Toolbar and a Pager. In this case, tab to move the focus from one nested component to another.

### Types of Keyboard Support

The Telerik UI for Blazor components may provide enhanced, standard, or no keyboard support. See the [compliance table]({%slug accessibility-compliance%}#compliance-table) for component-specific information.

> * *Standard keyboard support* implies similar keyboard navigation capabilities as standard HTML elements. For example, the Button components support `Enter` and `Space` for clicking them. All components with standard keyboard support are reachable through the `Tab` key and provide focus styles.

Reviewer: ```suggestion
* *Standard keyboard support* implies similar keyboard navigation capabilities as standard HTML elements. For example, the Button components support `Enter` and `Space` for triggering clicks. All components with standard keyboard support are reachable through the `Tab` key and provide focus styles.
```

You can [obtain and use the Default Ocean Blue A11y swatch]({%slug themes-overview%}#swatch) starting from the following component and theme versions:

* [Telerik UI for Blazor version 4.0.1](https://www.telerik.com/support/whats-new/blazor-ui/release-history/ui-for-blazor-4-0-1)
* [Themes version 6.0.3](https://github.com/telerik/kendo-themes/releases/tag/v6.0.3). Check section [Theme Version Compatibility]({%slug themes-overview%}#compatibility-and-maintenance) on how to align Telerik UI for Blazor versions with theme versions.

## Development Practices

When implementing a Telerik Blazor component, the team:

* Follows the WCAG standard and WAI-ARIA specification to lay the right foundation for the component accessibility.

> * Implements automated unit tests guarantee accessible and semantically correct rendering.

Reviewer: ```suggestion
* Implements automated unit tests to guarantee accessible and semantically correct rendering.
```

* Localizes messages for labels, titles, and other elements.

> * Manually tests the component to with regard to its keyboard navigation and usage with screen readers.

Reviewer: ```suggestion
* Manually tests the component with regard to its keyboard navigation and usage with screen readers.
```
